matRad_calcCubes.m

```diff
@@ -37,54 +37,72 @@
 resultGUI.w = w;
 
 % calc dose and reshape from 1D vector to 2D array
-resultGUI.physicalDose = reshape(full(dij.physicalDose{scenNum}*resultGUI.w),dij.dimensions);
+numOfBeams = max(unique(dij.beamNum));
+
+for i = 1:(numOfBeams + 1)
+    if i == (numOfBeams + 1)
+      beamInfo(i).suffix = '';
+      beamInfo(i).isCurrBeam = ones(size(dij.beamNum));
+    else
+      beamInfo(i).suffix = ['Beam', num2str(i)];
+      beamInfo(i).isCurrBeam = (dij.beamNum == i);      
+    end
+end
+
+for i = 1:length(beamInfo)
+    resultGUI.(['physicalDose', beamInfo(i).suffix]) = reshape(full(dij.physicalDose{scenNum} * (resultGUI.w .* beamInfo(i).isCurrBeam)),dij.dimensions);
+end
 
 % consider RBE for protons
 if isfield(dij,'RBE')
-   fprintf(['matRad: applying a constant RBE of ' num2str(dij.RBE) ' \n']); 
-   resultGUI.RBExDose     = resultGUI.physicalDose * dij.RBE;
+   fprintf(['matRad: applying a constant RBE of ' num2str(dij.RBE) ' \n']);
+   for i = 1:length(beamInfo)
+        resultGUI.(['RBExDose', beamInfo(i).suffix]) = resultGUI.(['physicalDose', beamInfo(i).suffix]) * dij.RBE;
+   end
 end
 
 % consider VOI priorities
 [cst,resultGUI.overlapCube]  = matRad_setOverlapPriorities(cst,dij.dimensions);
 
 if isfield(dij,'mLETDose')
-    LETDoseCube       = dij.mLETDose{scenNum} * resultGUI.w;
-    resultGUI.LET     = zeros(dij.dimensions);
-    ix                = resultGUI.physicalDose>0;
-    resultGUI.LET(ix) = LETDoseCube(ix)./resultGUI.physicalDose(ix);
-
+    for i = 1:length(beamInfo)
+        LETDoseCube                      = dij.mLETDose{scenNum} * (resultGUI.w .* beamInfo(i).isCurrBeam);
+        resultGUI.(['LET', beamInfo(i).suffix]) = zeros(dij.dimensions);
+        ix                               = resultGUI.(['physicalDose', beamInfo(i).suffix]) > 0;
+        resultGUI.(['LET', beamInfo(i).suffix])(ix) = LETDoseCube(ix)./resultGUI.(['physicalDose', beamInfo(i).suffix])(ix);
+    end
 end
 
 % consider biological optimization for carbon ions
 if isfield(dij,'mAlphaDose') && isfield(dij,'mSqrtBetaDose')
+    for i = 1:length(beamInfo)
 
-    a_x = zeros(size(resultGUI.physicalDose));
-    b_x = zeros(size(resultGUI.physicalDose));
+    a_x = zeros(size(resultGUI.(['physicalDose', beamInfo(i).suffix])));
```

Comment:
I think we should try to make all allocation processes independent of the arrays and better use

zeros(dij.dimensions);